bob/src/hw_metrics_collector.rs

```diff
@@ -157,30 +167,35 @@ impl HWMetricsCollector {
     // NOTE: HashMap contains only needed mount points of used disks, so it won't be really big,
     // but maybe it's more efficient to store disks (instead of mount_points) and update them one by one
 
-    fn space(disks: &HashMap<PathBuf, DiskName>) -> DiskSpaceMetrics {
-        let mut total = 0;
-        let mut used = 0;
-        let mut free = 0;
-
-        for mount_point in disks.keys() {
+    /// Maps [`DiskSpaceMetrics`] to the corresponding mount point.
+    ///
+    /// `f_fsid` field of [`statvfs`] is used to ensure that disks are unique.
+    fn space(disks: &HashMap<PathBuf, DiskName>) -> HashMap<PathBuf, DiskSpaceMetrics> {
+        let mut res = HashMap::new();
+        let mut fs_ids = HashSet::new();
+        for (mount_point, disk_name) in disks {
             let cm_p = Self::to_cpath(mount_point.as_path());
             let stat = Self::statvfs_wrap(&cm_p);
             if let Some(stat) = stat {
-                let bsize = stat.f_bsize as u64;
-                let blocks = stat.f_blocks as u64;
-                let bavail = stat.f_bavail as u64;
-                let bfree = stat.f_bfree as u64;
-                total += bsize * blocks;
-                free += bsize * bavail;
-                used += (blocks - bfree) * bsize;
+                let bsize = stat.f_bsize;
+                let blocks = stat.f_blocks;
+                let bavail = stat.f_bavail;
+                let bfree = stat.f_bfree;
```

Comment:
Please, return back explicit type declaration, because it is required for proper calculations. On some targets the type of all this values can be 'u32' instead of 'u64', so it is better to explicitly convert it to `u64` here to avoid overflow on multiplication operation later

Reply:
*nerd mode on*
*Theoretically*, if that's the case, it shouldn't even build at this point. Multiplication of two `u32` numbers produces another `u32` number, but  `DiskSpaceMetrics` has `u64` field, and rust prohibits **any** implicit type conversions, thus the compiler should produce the error
In order to convert the type we must use either `as <type>` syntax or `.into()` method
*nerd mode off*
But maybe I'm missing something

In any case, the type annotations are placed back, they wouldn't hurt